src/RAM/Science/components/rocklookup.tsx

```diff
@@ -0,0 +1,706 @@
+      const lineArr = textLine.split(/\t/)
+      rockTable.push({
+        name: lineArr[0],
+        minerals: lineArr[1].split("; "),
+        description: lineArr[2],
+        texture: lineArr[3].split("; ")
+      })
+
+    })
+    return rockTable
+  } else {
+    console.error("Failed loading Rock Table. File might not exist. Look in " + ROCKPATH)
+    return []
+  }
+}
+
+const MINPATH = path.join(__dirname, "../assets/rockLookupAssets/MineralDatabase.tsv")
+
+/**
+ * Loads the mineral data from the tsv defined by MINPATH
+ * @returns Map of minerals, empty map if database was not found
+ */
+function fillFrom_M_Database(): Map<string, Minerals> {
+  if (fs.existsSync(MINPATH)) {
+    let mineralTable: Map<string, Minerals> = new Map()
+    let filePull = fs
+      .readFileSync(MINPATH)
+      .toString()
+      .split(/\r\n|\n\r|\n|\r/)
+    filePull = filePull.slice(1, filePull.length) // gets rid of the sheet labels
+    filePull.forEach(textLine => {
+      const lineArr = textLine.split(/\t/)
+      mineralTable.set(lineArr[0], {
+        name: lineArr[0],
+        forms: lineArr[1].split("; "),
+        cleaveAndLuster: lineArr[2].split("; "),
+        colors: lineArr[3].split("; "),
+      })
+    })
+    return mineralTable
+  } else {
+    console.error("Failed loading Mineral Table. File might not exist. Look in " + MINPATH)
+    return new Map()
+  }
+}
+
+const MINARR: Map<string, Minerals> = fillFrom_M_Database()
+const ROCKARR: Rocks[] = fillFrom_R_Database()
+
+function populateColors(): string[] {
+  let colorList: string[] = []
+  MINARR.forEach((mineral: Minerals) => {
+    mineral.colors.map(color => {
+      colorList.push(color)
+    })
+  })
+  colorList.sort()
+  return [...new Set(colorList)]
+}
+
+const COLORMASTER: string[] = populateColors()
+
+function populateForms(): string[] {
+  let formList: string[] = []
+  MINARR.forEach((mineral: Minerals) => {
+    mineral.forms.map(form => {
+      formList.push(form)
+    })
+  })
+  formList.sort()
+  return [...new Set(formList)]
+}
+
+const FORMMASTER: string[] = populateForms()
+
+function populateCleaves(): string[] {
+  let cleaveList: string[] = []
+  MINARR.forEach((mineral: Minerals) => {
+    mineral.cleaveAndLuster.map(cleave => {
+      cleaveList.push(cleave)
+    })
+  })
+  cleaveList.sort()
+  return [...new Set(cleaveList)]
+}
+
+const CLEAVEMASTER: string[] = populateCleaves()
+
+function populateTextures(): string[] {
+  let TextureList: string[] = []
+  ROCKARR.forEach((rock: Rocks) => {
+    rock.texture.map(texture => {
+      if (texture != "") {
+        TextureList.push(texture)
+      }
+    })
+  })
+  TextureList.sort()
+  return [...new Set(TextureList)]
+}
+
+const TEXTUREMASTER: string[] = populateTextures()
+
+interface Minerals {
+  name: string
+  forms: string[]
+  cleaveAndLuster: string[]
+  colors: string[]
+}
+
+interface Rocks {
+  name: string
+  minerals: string[]
+  description: string
+  texture: string[]
```

Comment:
if there can be more than one texture, make this plural, otherwise make it not an array